src/Interactive/Features/Interactive/Core/InteractiveHost.LazyRemoteService.cs

```diff
@@ -62,20 +62,23 @@ public void Dispose(bool joinThreads)
                     }
 
                     bool initializing = true;
-                    cancellationToken.Register(() =>
-                    {
-                        if (initializing)
+                    RemoteExecutionResult initializationResult;
+                    using (cancellationToken.Register(
+                        () =>
```

Comment:
this seems... odd.  You're registering a cancellation callback, but you're immedaitely disposing?  I don't really understand what teh intent is here.  Can you comment the code?